lib/relp/relp_protocol.rb

```diff
@@ -1,87 +1,69 @@
 require 'relp/exceptions'
-require 'socket'
-require "openssl"
+require 'relp/version'
 
 module Relp
   class RelpProtocol
-    @@relp_version = '0'
-    # TODO: check whether this exact line is needed or if we can put in something
-    # custom to help debugging
-    @@relp_software = 'librelp,1.2.13,http://librelp.adiscon.com'
+    @@relp_version = '0'.freeze
+    @@relp_software = "ruby-relp,#{VERSION},https://github.com/ViaQ/Relp".freeze
+    @@required_commands = 'syslog'.freeze
 
-    def create_frame(txnr, command, message)
-      frame = {:txnr => txnr,
-               :command => command,
-               :message => message
-      }
-    end
-
-    def frame_write(socket, frame)
-      raw_data=[
-          frame[:txnr],
-          frame[:command],
-          frame[:message]
-      ].join(' ')
-      @logger.debug "Writing Frame #{frame.inspect}"
-      begin
-        socket.write(raw_data)
-      rescue Errno::EPIPE,IOError,Errno::ECONNRESET
-        raise Relp::ConnectionClosed
+    def frame_parse(content)
+      frame = Hash.new
+      if match = content.match(/\A([0-9]+) ([\S]*) (\d+)(?: ?)([\s\S]*)\z/)
+        frame[:txnr], frame[:command], frame[:data_length], frame[:message] = match.captures
+      else
+        raise Relp::FrameParseException.new("Invalid RELP frame: #{content}")
```

Comment:
ok - but 
- maybe this is a ruby idiom, but using regexp in a high performance protocol is not optimal - consider using something non-regexp based like rsyslog lognorm or grok

Reply:
Great idea as that would also make it possible to support a client that sent multiple messages at once. I had briefly considered it but was keeping it close to what was there before. If I see performance issues or find I need to make additional changes I will consider it.